### DISCUSSION TITLE
How to log errors and request body to a seperate log file while also sending a premade response to client? - FastAPI with uvicorn and gunicorn

### DISCUSSION THREAD

**Discussion by naveenmarthala** [fastapi/fastapi#6504] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Description and Example Code

What I am trying to achieve:  
I have a simple FastAPI CPU bound app, that takes in application/json and returns application/json. I am using pydantic's `BaseModel`s for input and output data validation and many input requests generate validation errors. I would like to log the request body and the error message to a seperate log file to be able to process them later to debug, along with the timestamp; while also sending a pre-made response to the client.

Here's how I am running gunicorn and tried to capture it:
```
$ gunicorn --woker-class uvicorn.workers.UvicornWorker -capture-output --log-level debug --error-log-file /home/dockerd/gunicorn_log.log api.main:app
```

Here's how i am handling validation errors and where i presume changes need to be made:  
```
import sys
from fastapi import FastAPI
# for exception handling
from fastapi import Request
from fastapi.exceptions import RequestValidationError, ValidationError
# to change the return data to custom format
from fastapi.responses import JSONResponse


# create an instance for API server
app = FastAPI()

# a very generic function that gets triggered if input doesn't match my BaseModel
@app.exception_handler(RequestValidationError)
@app.exception_handler(ValidationError)
async def http422_error_handler(request: Request, exc:Union[RequestValidationError, ValidationError]):
    ## return pre-made response to client.
    return JSONResponse(
        status_code=422,
        content={
            'message': 'check the input json.'
            }
        )
```

Some errors like, when https requests are sent to http api, they get printed to stderr, that uvicorn prints and these do get captured in file specified in gunicorn's "--error-log-file" flag. but, this file will also include other gunicorn's details like starting the server etc,.

So, while logging gunicorn's details to a log-file, how do I also track the error and the request body that led to the error into a separate log file with the timestamp?

### Operating System

Linux

### Operating System Details

debian-bullseye-slim inside a docker image.

### FastAPI Version

fastapi==0.71.0

### Python Version

Python 3.8.12

### Additional Context

no additional context.

**Comment by jgould22** [comment, marked as answer]
You can use pythons logging module to setup a logger that logs to specific file (https://docs.python.org/3/howto/logging.html#logging-basic-tutorial)

Once you have the above logger you can pass the config (file or dict) to gunicorn using https://docs.gunicorn.org/en/stable/settings.html#logconfig

Once you have your custom logger you can log to it using a custom route/ APIRoute class https://fastapi.tiangolo.com/advanced/custom-request-and-route/?h=logging#custom-request-and-apiroute-class
